yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -519,8 +528,7 @@ private[spark] class Client(
     ).foreach { case (flist, resType, addToClasspath) =>
       flist.foreach { file =>
         val (_, localizedPath) = distribute(file, resType = resType)
-        require(localizedPath != null)
-        if (addToClasspath) {
+        if (addToClasspath && localizedPath != null) {
```

Comment:
@andrewor14 , in the previous code we assume all the files will be uploaded into distributed cache, so this `localizedPath` should not be null. But here with my change, some duplicated files will be neglected, this is return `localizedPath` as null instead, so here I change to this way.